### PR TITLE
DOCS DOCS DOCS

### PR DESCRIPTION
**Pull request by cansavvy**
#### What changes are being implemented in this Pull Request?
We need to have this documented and have a vignette! That's what this PR is about. 

#### What was your approach?

Wrote how to use this!

#### What GitHub issue does your pull request address?
#30

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,195 @@
+<!-- README.md is generated from README.Rmd. Please edit that file -->
+
+# metricminer
+
+`metricminer` is an R package that helps you mine metrics on common places on the web through the power of their APIs.
+
+It also helps make the data in a format that is easily used for a dashboard or other purposes.
```
Comment (carriewright11) [inline README.md:7]:
```suggestion
It also helps format the data so that it can easily be used for a dashboard or other purposes.
```

Comment (carriewright11) [reply README.md:7]: just simplifying this

File: README.md [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+
+- [Calendly](https://calendly.com/)
+- [GitHub](https://github.com/)
+- [Google Analytics](https://analytics.google.com/analytics/academy/course/6)
+- [Google Forms](https://www.google.com/forms/about/)
+- [Slido](https://admin.sli.do/events) export files stored on Googledrive
+
+## Data format options
+
+`metricminer` attempts to retrieve API data for you and give you it to you in a format that is a tidy data.frame.
```
Comment (carriewright11) [inline README.md:24]:
```suggestion
`metricminer`  retrieves API data for you and gives it to you in a format that is a tidy data.frame.
```

File: README.md [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+this means metricminer has to be opinionated about what metrics it returns so it fits in a useful and human ready to read data frame.
```
Comment (carriewright11) [inline README.md:25]:
```suggestion
this means metricminer has to be opinionated about what metrics it returns so it fits in a useful data frame easily read by humans.
```

File: README.md [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+
+If you find that the data returned is not what you need you have two options (these options can be pursued concurrently):
+
+1. You can set the `dataformat` argument to `"raw"` to see the original, unedited JSON formatted data as it was returned from the API. Then you can personally look for the data that you want and extract it.
+2. You can post a GitHub issue to explain why the metric missing from the data frame formatted data should be included. And if possible and reasonable, we can work on including that data in the next version of `metricminer`.
```
Comment (carriewright11) [inline README.md:30]:
```suggestion
2. You can post a GitHub issue to explain why the metric missing from the data frame of formatted data should be included. And if possible and reasonable, we can work on including that data in the next version of `metricminer`.
```

File: README.md [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+
+In Google Analytics `metrics` are your basic numbers (how many visits to your website, etc.).
+```
+metrics <- get_ga_stats(property_id, stats_type = "metrics")
+```
+Whereas `dimensions` are more  a list of events that have happened. So here's a list of people that have logged on.
+```
+dimensions <- get_ga_stats(property_id, stats_type = "dimensions")
+```
+Lastly, we have a third option of collecting `link_clicks` and the links they have clicked. This is also known as a dimension according to Google analytics, but often it isn't compatible for us to download link click data at the same time as other dimension data so in `metricminer` we collect them separately.
```
Comment (carriewright11) [inline README.md:114]:
```suggestion
Lastly, we have a third option of collecting `link_clicks` and the links they have clicked. This is also known as a dimension according to Google Analytics, but often it isn't compatible for us to download link click data at the same time as other dimension data so in `metricminer` we collect them separately.
```

File: README.md [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+If you have used Slido for interactive slide sessions and collected that info and exported it to your googledrive you can use `metricminer` to collect that data as well.
+
+```
+drive_id <- "https://drive.google.com/drive/folders/0AJb5Zemj0AAkUk9PVA"
+slido_data <- get_slido_files(drive_id)
+```
+
+## Bulk Retrievals
+
+Maybe you just want to retrieval it ALL. We have som wrapper functions that will attempt to do this for you.
```
Comment (carriewright11) [inline README.md:139]:
```suggestion
Maybe you just want to retrieve it ALL. We have some wrapper functions that will attempt to do this for you.
```

File: README.md [hunk 7]
```diff
@@ -0,0 +1,195 @@
+<!-- README.md is generated from README.Rmd. Please edit that file -->
+
+# metricminer
```
[the remark this answers is not in the document]
Comment (carriewright11) [reply README.md:3]: woohoo! looks good - maybe we should link to our article about metrics? [arXiv:2306.03255](https://arxiv.org/abs/2306.03255)

[the remark this answers is not in the document]
Comment (cansavvy) [reply README.md:3]: Yes yes! For sure! I'll add

File: README.md (hunk 1 above)
Comment (cansavvy) [inline README.md:7]:
```suggestion
It also helps make the data in a format that is easily used for a dashboard or other purposes.
You can read more about [suggested techniques for metrics in our associated manuscript](https://arxiv.org/abs/2306.03255).
```

File: README.md [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+
+`metricminer` is an R package that helps you mine metrics on common places on the web through the power of their APIs.
+
+It also helps format the data so that it can easily be used for a dashboard or other purposes.
+It will have an associated [dashboard template](https://github.com/fhdsl/metricminer-dashboard) and tutorials to help you fully use the data you retrieve with `metricminer`  (but these are still under development!)
+
+You can [read the metricminer package documentation here](https://hutchdatascience.org/metricminer/).
```
Comment (cansavvy) [inline README.md:10]:
```suggestion
- You can [read the metricminer package documentation here](https://hutchdatascience.org/metricminer/).
- And you can read more about metric collection in our [associated manuscript -- currently a preprint](https://arxiv.org/abs/2306.03255). 
```

File: vignettes/getting-started.Rmd [hunk 9]
```diff
@@ -0,0 +1,320 @@
...
+```
+
+Now in your script if you run the following, you will have authorization to GitHub. 
+
+```
+# Authorize GitHub
+auth_from_secret("github", token = Sys.getenv("METRICMINER_GITHUB_PAT"))
+```
+
+#### Setting Google auth from secret
```
Comment (cansavvy) [inline vignettes/getting-started.Rmd:241]: This is the part you need for authorizing from secrets @howardbaek

File: vignettes/getting-started.Rmd [hunk 10]
```diff
@@ -0,0 +1,320 @@
...
+  shared_drive = googledrive::as_id("0AJb5Zemj0AAkUk9PVA"),
+  type = "form")
+```
+
+Now we can provide this vector of form ids to `get_multiple_forms`
+```
+multiple_forms <- get_multiple_forms(form_ids = form_list$id)
+```
+
+### Non-interactive authorizing from secrets
```
Comment (cansavvy) [inline vignettes/getting-started.Rmd:205]:
```suggestion
## Non-interactive authorizing from secrets
```

File: vignettes/getting-started.Rmd [hunk 11]
```diff
@@ -0,0 +1,320 @@
...
+
+If you'd like to authorize non-interactively (whether on GitHub actions or locally) you can set your tokens using `Sys.setenv()` 
+
+#### Setting Calendly auth from secret
```
Comment (cansavvy) [inline vignettes/getting-started.Rmd:209]:
```suggestion
### Setting Calendly auth from secret 
```

File: vignettes/getting-started.Rmd [hunk 12]
```diff
@@ -0,0 +1,320 @@
...
+```
+Sys.setenv(METRICMINER_CALENDLY = "Put calendly token here")
+```
+
+Now in your script if you run the following, you will have authorization to Calendly. 
+```
+auth_from_secret("calendly", token = Sys.getenv("METRICMINER_CALENDLY"))
+```
+
+#### Setting GitHub auth from secret
```
Comment (cansavvy) [inline vignettes/getting-started.Rmd:223]:
```suggestion
### Setting GitHub auth from secret 
```

File: vignettes/getting-started.Rmd (hunk 9 above)
Comment (howardbaek) [reply vignettes/getting-started.Rmd]: Thanks for pointing this out.

**Comment by cansavvy** [comment]
Thanks for the review @carriewright11 I'm going to merge this so the docs are seen by people who come to GitHub.

**Pull request merged**